### PR TITLE
Scott/439 test invalid pegin gets reverted

### PR DESCRIPTION
**Pull request by scottmillner**

### CODE REVIEW

File: crates/blockchain-tree/src/noop.rs [hunk 1]
```diff
@@ -52,6 +52,15 @@ impl BlockchainTreeEngine for NoopBlockchainTree {
         ))
     }
 
+    fn insert_block_with_botanix_consensus_package(
+        &self,
+        block: SealedBlockWithSenders,
+        validation_kind: BlockValidationKind,
+        _botanix_consensus_pkg: Option<BotanixConsensusPackage>,
```
Comment (arminsabouri) [inline crates/blockchain-tree/src/noop.rs:59]: Should this get passed into insert_block?

File: crates/consensus/beacon/src/engine/message.rs [hunk 2]
```diff
@@ -151,6 +152,8 @@ pub enum BeaconEngineMessage<Engine: EngineTypes> {
         cancun_fields: Option<CancunPayloadFields>,
         /// The sender for returning payload status result.
         tx: oneshot::Sender<Result<PayloadStatus, BeaconOnNewPayloadError>>,
+        /// Optional BotaniX consensus package.
```
Comment (arminsabouri) [inline crates/consensus/beacon/src/engine/message.rs]: Big X typo

File: crates/blockchain-tree/src/noop.rs (hunk 1 above)
Comment (arminsabouri) [reply crates/blockchain-tree/src/noop.rs:59]: Or should we be using `insert_block_with_botanix_consensus_package ` here ?

File: crates/interfaces/src/error.rs [hunk 3]
```diff
@@ -60,21 +60,21 @@ impl From<FsPathError> for RethError {
 }
 
 // Some types are used a lot. Make sure they don't unintentionally get bigger.
-#[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
-mod size_asserts {
-    use super::*;
+// #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
```
Comment (arminsabouri) [inline crates/interfaces/src/error.rs]: Is this worth re-enabling? Maybe ticket out so we dont forget?

**Review by arminsabouri — COMMENTED** [review]
Just had a few nits.
approach Ack! Nice work

File: crates/blockchain-tree/src/noop.rs (hunk 1 above)
Comment (scottmillner) [reply crates/blockchain-tree/src/noop.rs:59]: i added `insert_block_with_botanix_consensus_package` to `BlockchainTreeEngine` trait but in this implementation of the trait, it isn't needed. So i've made `insert_block_with_botanix_consensus_package` just a wrapper around `insert_block`...so basically this is the same as calling `insert_block`

This is the approach im taking anytime the method needs to be implemented only bc it's part of the trait but the functionality isn't actually needed by the struct...was a tradeoff off extending the trait

File: crates/consensus/beacon/src/engine/message.rs (hunk 2 above)
Comment (scottmillner) [reply crates/consensus/beacon/src/engine/message.rs]: thanks! it looks kinda cool tho with the big X 😆... and done!

File: crates/interfaces/src/error.rs (hunk 3 above)
Comment (scottmillner) [reply crates/interfaces/src/error.rs]: i had to disable this bc `static_assert_size!(BlockExecutionError, 56);` fails bc the error message ends up being too big bc i need to pass pegin data in the error

i can try and renable and increase the the allowed size just for `static_assert_size!(BlockExecutionError, 56);`

and done!

File: bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    // status should be 0 (failure)
+    it_info_print!("Pegin Tx Receipt", tx_receipt);
+    assert!(tx_receipt.status.unwrap().is_zero());
+
+    // pegin address balance after pegin
+    let pegin_address_final_balance =
+        botanix_eth_client.get_botanix_balance(eth_pegin_address.as_str()).await.unwrap();
+    it_info_print!("Final pegin address balance", pegin_address_final_balance);
+
+    assert_eq!(pegin_address_initial_balance, pegin_address_final_balance);
```
Comment (arminsabouri) [inline bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs:223]: Good assertion! Is there anything else to assert about this account that would indicate a revert happened? Like nonce is different?

File: bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegout.rs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    let pegout_amount = Amount::from_btc(0.5).unwrap();
+    let tx_receipt = botanix_eth_client
+        .burn(invalid_pegout_destination, pegout_data, pegout_amount.to_wei())
+        .await
+        .unwrap()
+        .unwrap();
+    it_info_print!("Pegout Tx Receipt: ", tx_receipt);
+
+    assert!(tx_receipt.status.unwrap().is_zero());
+
```
Comment (arminsabouri) [inline bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegout.rs:58]: Maybe we assert value is the same as before for this account?

Comment (scottmillner) [reply bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegout.rs:58]: here im using an invalid address so there will be no balance

File: bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs (hunk 4 above)
Comment (scottmillner) [reply bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs:223]: good idea, done!

**Review by arminsabouri — APPROVED** [review]

**Pull request merged**